clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -502,14 +502,8 @@ private void doSend(ClientRequest clientRequest, boolean isInternalRequest, long
         String destination = clientRequest.destination();
         RequestHeader header = clientRequest.makeHeader(request.version());
         if (log.isDebugEnabled()) {
-            int latestClientVersion = clientRequest.apiKey().latestVersion();
-            if (header.apiVersion() == latestClientVersion) {
-                log.trace("Sending {} {} with correlation id {} to node {}", clientRequest.apiKey(), request,
```

Comment:
Given that we have to work with all client versions, it's just as common for the client not to match the broker version, so it's not really useful for the behavior to be different when they do match.

Reply:
SG.